src/tests/fail_foldRight.cpp

```diff
@@ -0,0 +1,14 @@
+#include <vector>
+#include <iostream>
+#include "../collections.h"
+
+
+int main() {
+    Collection<int> ints = Collection<int>(std::vector<int> {2, 3, 4, 5});
+
```

Comment:
Can you add a comment detailing why this should fail?
It'll make things easier to maintain